### ISSUE TITLE
Bug: Error loading script: train_tools.py

### ISSUE THREAD

**Issue by YuK1Game** [liasece/sd-webui-train-tools#9]
When I try to install, the following error message appears.

```
Error loading script: train_tools.py
Traceback (most recent call last):
  File "D:\Data\AI\stable-diffusion-webui\modules\scripts.py", line 256, in load_scripts
    script_module = script_loading.load_module(scriptfile.path)
  File "D:\Data\AI\stable-diffusion-webui\modules\script_loading.py", line 11, in load_module
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\Data\AI\stable-diffusion-webui\extensions\sd-webui-train-tools\scripts\train_tools.py", line 7, in <module>
    from liasece_sd_webui_train_tools.ui import *
  File "D:\Data\AI\stable-diffusion-webui\extensions\sd-webui-train-tools\liasece_sd_webui_train_tools\ui.py", line 16, in <module>
    from liasece_sd_webui_train_tools.train_ui import *
  File "D:\Data\AI\stable-diffusion-webui\extensions\sd-webui-train-tools\liasece_sd_webui_train_tools\train_ui.py", line 10, in <module>
    from liasece_sd_webui_train_tools import train
  File "D:\Data\AI\stable-diffusion-webui\extensions\sd-webui-train-tools\liasece_sd_webui_train_tools\train.py", line 11, in <module>
    from liasece_sd_webui_train_tools.Parser import Parser, ensure_path
```

Python version: 3.10.6

**Comment by liasece** [comment]
This does not seem to be the complete message. Please complete the error message.

**Comment by z-funfan** [comment]
Same problem on MAC Mini M1 Chip,
Python 3.10.11
Restart webui after extension installed, raised error message belows:

`
################################################################
Repo already cloned, using it as install directory
################################################################

################################################################
Create and activate python venv
################################################################

################################################################
Launching launch.py...
################################################################
Python 3.10.11 (main, Apr 19 2023, 23:03:08) [Clang 12.0.5 (clang-1205.0.22.9)]
Commit hash: 22bcc7be428c94e9408f589966c2040187245d81
Installing requirements for Web UI
Error running install.py for extension /Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools.
Command: "/Users/username/Project/github/stable-diffusion-webui/venv/bin/python3.10" "/Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/install.py"
Error code: 1
stdout: Installing sd-webui-train-tools requirement: /Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/requirements.txt

stderr: Traceback (most recent call last):
  File "/Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/install.py", line 15, in <module>
    launch.run_pip(f"install -r {python_requirements_file}", f"sd-webui-train-tools requirement: {python_requirements_file}")
  File "/Users/username/Project/github/stable-diffusion-webui/launch.py", line 129, in run_pip
    return run(f'"{python}" -m pip {args} --prefer-binary{index_url_line}', desc=f"Installing {desc}", errdesc=f"Couldn't install {desc}")
  File "/Users/username/Project/github/stable-diffusion-webui/launch.py", line 97, in run
    raise RuntimeError(message)
RuntimeError: Couldn't install sd-webui-train-tools requirement: /Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/requirements.txt.
Command: "/Users/username/Project/github/stable-diffusion-webui/venv/bin/python3.10" -m pip install -r /Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/requirements.txt --prefer-binary
Error code: 1
stdout: Looking in indexes: https://mirrors.aliyun.com/pypi/simple
Requirement already satisfied: accelerate in ./venv/lib/python3.10/site-packages (from -r /Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/requirements.txt (line 1)) (0.12.0)
Requirement already satisfied: transformers in ./venv/lib/python3.10/site-packages (from -r /Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/requirements.txt (line 2)) (4.25.1)
Requirement already satisfied: ftfy in ./venv/lib/python3.10/site-packages (from -r /Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/requirements.txt (line 3)) (6.1.1)
Collecting albumentations
  Using cached https://mirrors.aliyun.com/pypi/packages/4f/55/3c2ce84c108fc1d422afd6de153e4b0a3e6f96ecec4cb9afcf0284ce3538/albumentations-1.3.0-py3-none-any.whl (123 kB)
Requirement already satisfied: opencv-python in ./venv/lib/python3.10/site-packages (from -r /Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/requirements.txt (line 5)) (4.7.0.72)
Requirement already satisfied: einops in ./venv/lib/python3.10/site-packages (from -r /Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/requirements.txt (line 6)) (0.4.1)
Collecting diffusers[torch]==0.10.2
  Using cached https://mirrors.aliyun.com/pypi/packages/5d/44/196b5c8a29aad00c8b68d71ba0712a75b1e727d5ee7fcd32a3657270c628/diffusers-0.10.2-py3-none-any.whl (503 kB)
Requirement already satisfied: pytorch-lightning in ./venv/lib/python3.10/site-packages (from -r /Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/requirements.txt (line 8)) (1.9.4)
Collecting bitsandbytes
  Using cached https://mirrors.aliyun.com/pypi/packages/0d/95/04de4035b1874026cadb9219aafb890e14fc6f3702a67618a144c66ec710/bitsandbytes-0.38.1-py3-none-any.whl (104.3 MB)
Collecting tensorboard
  Using cached https://mirrors.aliyun.com/pypi/packages/aa/80/f7233129f75d0d1b35e67df3a48010fffd21ccde124847e3c33d503fef01/tensorboard-2.12.2-py3-none-any.whl (5.6 MB)
Requirement already satisfied: safetensors in ./venv/lib/python3.10/site-packages (from -r /Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/requirements.txt (line 11)) (0.3.0)
Requirement already satisfied: gradio in ./venv/lib/python3.10/site-packages (from -r /Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/requirements.txt (line 12)) (3.23.0)
Requirement already satisfied: altair in ./venv/lib/python3.10/site-packages (from -r /Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/requirements.txt (line 13)) (4.2.2)
Collecting easygui
  Using cached https://mirrors.aliyun.com/pypi/packages/8e/a7/b276ff776533b423710a285c8168b52551cb2ab0855443131fdc7fd8c16f/easygui-0.98.3-py2.py3-none-any.whl (92 kB)
Requirement already satisfied: requests in ./venv/lib/python3.10/site-packages (from -r /Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/requirements.txt (line 16)) (2.25.1)
Requirement already satisfied: timm in ./venv/lib/python3.10/site-packages (from -r /Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/requirements.txt (line 17)) (0.6.7)
Collecting fairscale
  Using cached fairscale-0.4.13-py3-none-any.whl

stderr: ERROR: Could not find a version that satisfies the requirement tensorflow (from versions: none)
ERROR: No matching distribution found for tensorflow

[notice] A new release of pip is available: 23.0.1 -> 23.1
[notice] To update, run: pip install --upgrade pip





Launching Web UI with arguments: --skip-torch-cuda-test --upcast-sampling --no-half-vae --medvram
Warning: caught exception 'Torch not compiled with CUDA enabled', memory monitor disabled
No module 'xformers'. Proceeding without it.
==============================================================================
You are running torch 1.12.1.
The program is tested to work with torch 1.13.1.
To reinstall the desired version, run with commandline flag --reinstall-torch.
Beware that this will cause a lot of large files to be downloaded, as well as
there are reports of issues with training tab on the latest version.

Use --skip-version-check commandline argument to disable this check.
==============================================================================
Error loading script: train_tools.py
Traceback (most recent call last):
  File "/Users/username/Project/github/stable-diffusion-webui/modules/scripts.py", line 256, in load_scripts
    script_module = script_loading.load_module(scriptfile.path)
  File "/Users/username/Project/github/stable-diffusion-webui/modules/script_loading.py", line 11, in load_module
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/scripts/train_tools.py", line 7, in <module>
    from liasece_sd_webui_train_tools.ui import *
  File "/Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/liasece_sd_webui_train_tools/ui.py", line 16, in <module>
    from liasece_sd_webui_train_tools.train_ui import *
  File "/Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/liasece_sd_webui_train_tools/train_ui.py", line 10, in <module>
    from liasece_sd_webui_train_tools import train
  File "/Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/liasece_sd_webui_train_tools/train.py", line 8, in <module>
    from liasece_sd_webui_train_tools.ArgsList import ArgStore
  File "/Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/liasece_sd_webui_train_tools/ArgsList.py", line 22, in <module>
    import liasece_sd_webui_train_tools.sd_scripts.train_network as train_network
  File "/Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/liasece_sd_webui_train_tools/sd_scripts/train_network.py", line 10, in <module>
    import toml
ModuleNotFoundError: No module named 'toml'

Loading weights [fc2511737a] from /Users/username/Project/github/stable-diffusion-webui/models/Stable-diffusion/真实系：chilloutmix_NiPrunedFp32Fix.safetensors
Creating model from config: /Users/username/Project/github/stable-diffusion-webui/configs/v1-inference.yaml
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
`

Then I run `pip install toml`, it tells me toml already installed.
`Requirement already satisfied: toml in /Users/username/.pyenv/versions/3.10.11/lib/python3.10/site-packages (0.10.2)`

**Comment by liasece** [comment]
> ################################################################ Launching launch.py... 
> ################################################################
> Create and activate python venv
> ################################################################
> Error loading script: train_tools.py Traceback (most recent call last): File "/Users/username/Project/github/stable-diffusion-webui/modules/scripts.py", line 256, in load_scripts script_module = script_loading.load_module(scriptfile.path) File "/Users/username/Project/github/stable-diffusion-webui/modules/script_loading.py", line 11, in load_module module_spec.loader.exec_module(module) File "", line 883, in exec_module File "", line 241, in _call_with_frames_removed File "/Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/scripts/train_tools.py", line 7, in from liasece_sd_webui_train_tools.ui import * File "/Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/liasece_sd_webui_train_tools/ui.py", line 16, in from liasece_sd_webui_train_tools.train_ui import * File "/Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/liasece_sd_webui_train_tools/train_ui.py", line 10, in from liasece_sd_webui_train_tools import train File "/Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/liasece_sd_webui_train_tools/train.py", line 8, in from liasece_sd_webui_train_tools.ArgsList import ArgStore File "/Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/liasece_sd_webui_train_tools/ArgsList.py", line 22, in import liasece_sd_webui_train_tools.sd_scripts.train_network as train_network File "/Users/username/Project/github/stable-diffusion-webui/extensions/sd-webui-train-tools/liasece_sd_webui_train_tools/sd_scripts/train_network.py", line 10, in import toml ModuleNotFoundError: No module named 'toml'
> 
> Then I run `pip install toml`, it tells me toml already installed. `Requirement already satisfied: toml in /Users/username/.pyenv/versions/3.10.11/lib/python3.10/site-packages (0.10.2)`

If you are using python venv, you should run `venv/Scripts/pip install toml ` instead of pip.
